spec/models/challenge_spec.rb

```diff
@@ -236,8 +236,10 @@
 
         specify { expect(described_class.viewable_by_user(user_1)).to include(challenge) }
         specify { expect(described_class.viewable_by_user(user_1).count).to eq(1) }
+        specify { expect(described_class.viewable_by_user(user_1).first.user_challenges.length).to eq(challenge.user_challenges.count) }
```

Comment:
Question, does the data you set up here reflect cross row challenges? Like someone in A row challenging a spot in X row and users 1 and 2 are squad leaders in A and X row?

Reply:
Yes, but it kinda relies on the challenge factory not changing. The factory makes a user in row a and a user in the spot of the challenge, so I set the spot to be in x row:

```ruby
[2] pry(#<RSpec::ExampleGroups::Challenge::ViewableByUser::WhenASquadLeaderIsEvaluating::AndTheyAreInTheRowOfOneOfTheParticipants>)> challenge.user_challenges.map(&:spot)
=> [#<Spot:0x007fa823443c98 id: 129, row: "a", file: 13, created_at: Thu, 07 Sep 2017 01:31:18 UTC +00:00, updated_at: Thu, 07 Sep 2017 01:31:18 UTC +00:00>,
 #<Spot:0x007fa82337b6d0 id: 128, row: "x", file: 1, created_at: Thu, 07 Sep 2017 01:31:18 UTC +00:00, updated_at: Thu, 07 Sep 2017 01:31:18 UTC +00:00>]
```

Reply:
[Here is the code for the factory](https://github.com/osumb/challenges/blob/master/spec/factories/challenges.rb#L33-L34)